### PR TITLE
added dynamic ssl policy

### PR DESCRIPTION
**Pull request by cpareek**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSPO-13210

### Change description ###

Added `ssl_policy` dynamic block so that we can choose which `policy_type` and `protocol_version` we can use, by default the variable would be `null`. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -191,6 +191,18 @@ resource "azurerm_application_gateway" "ag" {
     }
   }
 
+  dynamic "ssl_policy" {
```
Comment (thomast1906) [inline main.tf:194]: Is this written the simplest way?

File: main.tf [hunk 2]
```diff
@@ -191,6 +191,18 @@ resource "azurerm_application_gateway" "ag" {
...
+    for_each = var.ssl_policy != null ? [var.ssl_policy] : []
+    content {
+      disabled_protocols   = var.ssl_policy.policy_type == null && var.ssl_policy.policy_name == null ? var.ssl_policy.disabled_protocols : null
```
Comment (thomast1906) [inline main.tf:197]: are all nulls needed if the top-level dynamic is null?

File: main.tf (hunk 1 above)
Comment (cpareek) [reply main.tf:194]: yupp - I have followed same dynamic blocks logic which are already added on the `azurerm_application_gateway` resource :)

File: main.tf (hunk 2 above)
Comment (cpareek) [reply main.tf:197]: Yes - if we do not have default null for objects, we will have to pass something from the calling repository. This way, we can only pass value of the `policy_name` and `min_protocol_version`  from calling repo.

**Review by TomKimber — APPROVED** [review]

**Pull request merged**